src/forms/superfluidMinionProposal.js

```diff
@@ -0,0 +1,274 @@
+    }
+  }, [errors]);
+
+  const onSubmit = async (values) => {
+    setLoading(true);
+
+    const valueWei = injectedProvider.utils.toWei(values.paymentRequested);
+    const details = detailsToJSON({
+      title: values.title,
+      description: values.description,
+      link: values.link,
+      recipient: values.memberApplicant || values.applicant,
+      token: values.paymentToken,
+      tokenRate: `${values.tokenRate} ${values.baseRate}`,
+      ratePerSec: values.weiRatePerSec,
+      minDeposit: valueWei,
+    });
+    const args = [
+      values.memberApplicant || values.applicant,
+      values.paymentToken,
+      values.weiRatePerSec,
+      valueWei,
+      '0x',
+      details,
+    ];
+
+    try {
+      const poll = createPoll({ action: 'superfluidProposeAction', cachePoll })(
+        {
+          minionAddress: values.minionContract,
+          createdAt: now,
+          chainID: daochain,
+          actions: {
+            onError: (error, txHash) => {
+              errorToast({
+                title: `There was an error.`,
+              });
+              resolvePoll(txHash);
+              console.error(`Could not find a matching proposal: ${error}`);
+            },
+            onSuccess: (txHash) => {
+              successToast({
+                title: 'Minion proposal submitted.',
+              });
+              refreshDao();
+              resolvePoll(txHash);
+            },
+          },
+        },
+      );
+      const onTxHash = () => {
+        setProposalModal(false);
+        setTxInfoModal(true);
+      };
+      await SuperfluidMinionService({
+        web3: injectedProvider,
+        minion: values.minionContract,
+        chainID: daochain,
+      })('proposeStream')({ args, address, poll, onTxHash });
+    } catch (err) {
+      setLoading(false);
+      console.log('error: ', err);
+    }
+  };
+
+  watch('paymentToken', '');
+
+  return minions.length ? (
+    <form onSubmit={handleSubmit(onSubmit)}>
+      <FormControl
+        isInvalid={errors.name}
+        display='flex'
+        flexDirection='row'
+        justifyContent='space-between'
+        mb={5}
+        flexWrap='wrap'
+      >
+        <Box w={['100%', null, '50%']} pr={[0, null, 5]}>
+          <Stack spacing={2}>
+            <Box>
+              <TextBox as={FormLabel} size='xs' htmlFor='minionContract'>
+                Minion Contract
+              </TextBox>
+              <Select
+                name='minionContract'
+                icon={<AiOutlineCaretDown />}
+                mb={5}
+                focusBorderColor='secondary.500'
+                ref={register({
+                  required: {
+                    value: true,
+                    message: 'Superfluid Minion contract is required',
+                  },
+                })}
+                placeholder='Select A Superfluid Agreement'
+              >
+                {' '}
+                {minions.map((minion, idx) => (
```

Comment:
Using index for key confuses the react diffing engine. We need to use a value that's unique.